monai/transforms/io/array.py

```diff
@@ -514,3 +515,46 @@ def __call__(
             "    https://docs.monai.io/en/latest/installation.html#installing-the-recommended-dependencies.\n"
             f"   The current registered writers for {self.output_ext}: {self.writers}.\n{msg}"
         )
+
+
+class MappingJson(Transform):
+    """
+    Writes a JSON file that logs the mapping between input image paths and their corresponding output paths.
+
+    Args:
+        mapping_json_path (Path or str): Path to the JSON file where the mappings will be saved.
+    """
+
+    def __init__(self, mapping_json_path: Path | str = "mapping.json"):
+        self.mapping_json_path = Path(mapping_json_path)
+
+    def write_json(self, input_path: str, output_path: str):
+        """
+        Args:
+            input_path (str): The path of the input image file.
+            output_path (str): The path of the output image file.
+        """
+        log_data = {"input": input_path, "output": output_path}
+        try:
+            with self.mapping_json_path.open("r") as f:
```

Comment:
I think this part is unsafe for the multiprocess environment. Could you please add a multiprocess test case reading/writing the same file here? If the test turns out failure, I think we should consider to use the [filelock](https://py-filelock.readthedocs.io/en/latest/index.html) to make this one safe. @ericspod  Please correct me if I am wrong about this one.

Thanks  
Bin

Reply:
If nothing is done to either ensure the file names are unique for every thread/process then yes writing to a file by multiple processes is going to be unsafe. Very fast writes can usually be done atomically so you don't often see the issue, but they are still unsafe. 

The solution is to ensure files are locked when written to, this requires a third party library like filelock (which seems well supported so we could consider adding it to requirements), writing to a database like sqlite, or writing to unique files that can be combined later. 

My thought on how output transforms would be used was that there would be a single thread/process doing the IO and so there would be no race issues, or if multiples they'd be guaranteed to write to unique files. This issue here of wanting to accumulate a JSON file from multiple processes is unique I think in MONAI.

Reply:
Hi, @binliunls @ericspod . I have added a multiprocess test case. The original `MappingJson` indeed results in failures. Therefore, I added filelock. What do you think about the new one?